src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -83,7 +90,7 @@ protected void performPoll() {
     }
 
     @Override
-    public void stop() {
+    public synchronized void stop() {
```

Comment:
Comment added:
    // Based on https://docs.confluent.io/platform/current/connect/devguide.html#task-example-source-task
    // This method needs to be synchronized "because SourceTasks are given a dedicated thread which they can block 
    // indefinitely, so they need to be stopped with a call from a different thread in the Worker."